### PR TITLE
feat(ddm): Links to release and transaction + summary and focus polishes

### LINKED ISSUE #56999: Polish summary table (hover states, series focus,etc.)

### LINKED ISSUE #57000: Add transaction/release links to the summary table (preserve time and other selections)

### LINKED ISSUE #57001: Hover over a single time series out of a group by and have it highlighted

### PR DESCRIPTION
**Pull request by matejminar**
This PR:
- polishes summary table (hover states, series focus, etc.)
- adds transaction+release links to the summary table
- makes it so that hovering over a single series focuses it

Closes https://github.com/getsentry/sentry/issues/56999
Closes https://github.com/getsentry/sentry/issues/57000
Closes https://github.com/getsentry/sentry/issues/57001

### CODE REVIEW

**Review by iambriccardo — COMMENTED** [review]
LGTM, will wait for @ArthurKnaus's final opinion.

File: static/app/views/ddm/summaryTable.tsx [hunk 1]
```diff
@@ -1,47 +1,127 @@
...
+                                environment: selection.environments,
+                                start: selection.datetime.start
+                                  ? getUtcDateString(selection.datetime.start)
+                                  : undefined,
+                                end: selection.datetime.end
+                                  ? getUtcDateString(selection.datetime.end)
+                                  : undefined,
+                                statsPeriod: selection.datetime.period,
+                              },
+                            })}
```
Comment (ArthurKnaus) [inline static/app/views/ddm/summaryTable.tsx:90]: small nit: maybe calculate this and the other `to` prop in the function body to keep the jsx easy to read?

**Review by ArthurKnaus — APPROVED** [review]
👍

File: static/app/views/ddm/summaryTable.tsx (hunk 1 above)
Comment (matejminar) [reply static/app/views/ddm/summaryTable.tsx:90]: @ArthurKnaus auto-merge kicked in, will apply the feedback in a follow-up 👍

**Pull request merged**